src/main/java/org/sagebionetworks/bridge/workers/dynamodb/streams/StreamRecordProcessor.java

```diff
@@ -0,0 +1,83 @@
+package org.sagebionetworks.bridge.workers.dynamodb.streams;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+
+import com.amazonaws.services.dynamodbv2.streamsadapter.model.RecordAdapter;
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.IRecordProcessor;
+import com.amazonaws.services.kinesis.clientlibrary.interfaces.IRecordProcessorCheckpointer;
+import com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason;
+import com.amazonaws.services.kinesis.model.Record;
+
+abstract class StreamRecordProcessor implements IRecordProcessor {
+
+    private final int checkpointInterval;
+
+    StreamRecordProcessor(final int checkpointInterval) {
+        this.checkpointInterval = checkpointInterval;
+    }
+
+    @Override
+    public void initialize(final String shardId) {
+        log().info("[ShardID=" + shardId + "]");
+    }
+
+    @Override
+    public void processRecords(final List<Record> records, final IRecordProcessorCheckpointer checkpointer) {
+        int count = 0;
+        for (final Record record : records) {
+            if(record instanceof RecordAdapter) {
+                final com.amazonaws.services.dynamodbv2.model.Record streamRecord
+                        = ((RecordAdapter) record).getInternalObject();
+                log().info("[EventId=" + streamRecord.getEventID() + ", " +
+                        "SequenceNumber=" + streamRecord.getDynamodb().getSequenceNumber() + "]");
+                process(streamRecord);
+                count += 1;
+                if (count % checkpointInterval == 0) {
+                    try {
+                        checkpointer.checkpoint();
+                    } catch(Exception e) {
+                        throw new RuntimeException(e);
+                    }
```

Comment:
Instead of catching all Exceptions, we should only catch the exceptions checkpointer.checkpoint() declares (or RuntimeException if it doesn't declare any).

Similarly, instead of throwing RuntimeException (which in tests, can easily be mixed up with a NullPointerException due to a bug in the code), we should throw a more specific exception or create our own RuntimeException to throw.